owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/MainFileListFragment.kt

```diff
@@ -206,6 +213,22 @@ class MainFileListFragment : Fragment(),
     private fun initViews() {
         setHasOptionsMenu(true)
         statusBarColorActionMode = ContextCompat.getColor(requireContext(), R.color.action_mode_status_bar_background)
+        dialog=BottomSheetDialog(requireContext())
+        val uploadBottomSheet = layoutInflater.inflate(R.layout.upload_bottom_sheet_fragment,
+            null)
+        dialog!!.setContentView(uploadBottomSheet)
+
+        uploadFromFilesItemView = uploadBottomSheet.findViewById(R.id.upload_from_files_item_view)
+        uploadFromCameraItemView =uploadBottomSheet.findViewById(R.id.upload_from_camera_item_view)
+        uploadToTextView = uploadBottomSheet.findViewById(R.id.upload_to_text_view)
+
+        val uploadBottomSheetBehavior: BottomSheetBehavior<*> = BottomSheetBehavior.from(uploadBottomSheet.parent as View)
+        dialog!!.setOnShowListener { uploadBottomSheetBehavior.setPeekHeight(uploadBottomSheet.measuredHeight) }
+
+        uploadToTextView?.text = String.format(
+            resources.getString(R.string.upload_to),
+            resources.getString(R.string.app_name)
+        )
```

Comment:
Here's the reason behind my logic: When I tested it on Pixel 3 api 33, if a user opens the dialog and rotates the device, the dialog was leaking. This is because in the initial code, the `dialog.show()` attempts to show the dialog even after the fragment has been destroyed/disposed. Below is the stack trace.

```
   android.view.WindowLeaked: Activity com.owncloud.android.ui.activity.FileDisplayActivity has leaked window DecorView@1dc9972[FileDisplayActivity] that was originally added here
                                                                                                    	at android.view.ViewRootImpl.<init>(ViewRootImpl.java:519)
                                                                                                    	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:346)
                                                                                                    	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
                                                                                                    	at android.app.Dialog.show(Dialog.java:329)
                                                                                                    	at com.owncloud.android.presentation.files.filelist.MainFileListFragment.openBottomSheetToUploadFiles(MainFileListFragment.kt:773)
                                                                                                    	at com.owncloud.android.presentation.files.filelist.MainFileListFragment.registerFabUploadListener$lambda$7(MainFileListFragment.kt:720)
                                                                                                    	at com.owncloud.android.presentation.files.filelist.MainFileListFragment.$r8$lambda$p4LwSRqoxddS_yXHPAsop8FOeG8(Unknown Source:0)
                                                                                                    	at com.owncloud.android.presentation.files.filelist.MainFileListFragment$$ExternalSyntheticLambda8.onClick(Unknown Source:2)
                                                                                                    	at android.view.View.performClick(View.java:6603)
                                                                                                    	at android.view.View.performClickInternal(View.java:6576)
                                                                                                    	at android.view.View.access$3100(View.java:780)
                                                                                                    	at android.view.View$PerformClick.run(View.java:26088)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:873)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loop(Looper.java:193)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:6702)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:911)
```

So the solution I could think of at the moment is making the `dialog` property a global null variable that is instatiated inside `onViewCreated()`, and the global variable can be whenever it is needed. The downside of this of cos, is as you've said, the dialog will be added to the view-tree despite it not being used but I think it is a fair trade off considering the leak. Please let me know what you think about this. Thanks